### PR TITLE
feat(apps): add RStudio

### PR DESCRIPTION
**Pull request by floriansipp**
Adapted the image from [rstudio hip](https://github.com/HIP-infrastructure/rstudio/blob/master/Dockerfile) 
Build time is a little bit more than 45mn so plan a café and a steak in beetween. 
Other than that seems to work nicely on ubuntu24

### CODE REVIEW

File: apps/rstudio/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    software-properties-common dirmngr \
+    curl wget gpg gpg-agent libssl-dev libpq5 libclang-dev \
+    libnss3 libatk1.0-0 libatk-bridge2.0-0 libcups2 \
+    libgtk-3-0 libasound2t64 build-essential cmake file \
+    gfortran liblapack-dev libopenblas-dev libpng-dev \
+    libxkbcommon-x11-0 && \
+    wget -qO- https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | tee -a /etc/apt/trusted.gpg.d/cran_ubuntu_key.asc && \
+    add-apt-repository "deb https://cloud.r-project.org/bin/linux/ubuntu $(lsb_release -cs)-cran40/" && \
```
Comment (greut) [inline apps/rstudio/Dockerfile:22]: that's the deprecated way of doing this.

Use `gpg` and `deb [signed-by=...]` like the other apps.

File: apps/rstudio/build.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
```
Comment (greut) [inline apps/rstudio/build.sh:2]:
```suggestion

set -e

```

File: apps/rstudio/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    apt-get install --no-install-recommends -y r-base r-cran-rstan r-cran-tidyverse && \
+    curl -sSO https://download1.rstudio.org/electron/jammy/amd64/rstudio-${APP_VERSION}-375-amd64.deb && \
```
Comment (greut) [inline apps/rstudio/Dockerfile:24]: 24.04 is not jammy.

Comment (floriansipp) [reply apps/rstudio/Dockerfile:24]: Good point but it's the link they put for all ubuntu versions of the official download page (and after checking this : https://download1.rstudio.org/ , no noble something) so who am i to argue ?

File: apps/rstudio/Dockerfile (hunk 1 above)
Comment (floriansipp) [reply apps/rstudio/Dockerfile:22]: Fair enough, will confirm back after coffee and steak

File: apps/rstudio/Dockerfile (hunk 3 above)
Comment (greut) [reply apps/rstudio/Dockerfile:24]: it's the same indeed. All good I guess.

Screenshot 2024-09-24 at 16-49-33 RStudio Desktop - Posit

File: apps/rstudio/Dockerfile (hunk 1 above)
Comment (floriansipp) [reply apps/rstudio/Dockerfile]: Should be good now

File: apps/rstudio/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    curl wget gpg gpg-agent software-properties-common \
+    build-essential cmake dirmngr file libssl-dev libpq5 \
+    libclang-dev libnss3 libatk1.0-0 libatk-bridge2.0-0 \
+    libcups2 libgtk-3-0 libasound2t64 gfortran liblapack-dev \
+    libopenblas-dev libpng-dev libxkbcommon-x11-0 && \
+    wget -qO- https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | gpg --dearmor > cran-archive-keyring.gpg && \
+    install -D -o root -g root -m 644 cran-archive-keyring.gpg /etc/apt/keyrings/cran-archive-keyring.gpg && \
+    sh -c 'echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/cran-archive-keyring.gpg] https://cloud.r-project.org/bin/linux/ubuntu $(lsb_release -cs)-cran40/" > /etc/apt/sources.list.d/cran.list' && \
```
Comment (greut) [inline apps/rstudio/Dockerfile:22]: I don't think you need to `sh -c` that.

File: apps/rstudio/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    curl wget gpg gpg-agent software-properties-common \
+    build-essential cmake dirmngr file libssl-dev libpq5 \
+    libclang-dev libnss3 libatk1.0-0 libatk-bridge2.0-0 \
+    libcups2 libgtk-3-0 libasound2t64 gfortran liblapack-dev \
+    libopenblas-dev libpng-dev libxkbcommon-x11-0 && \
+    wget -qO- https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | gpg --dearmor > cran-archive-keyring.gpg && \
+    install -D -o root -g root -m 644 cran-archive-keyring.gpg /etc/apt/keyrings/cran-archive-keyring.gpg && \
```
Comment (greut) [inline apps/rstudio/Dockerfile:21]: what's wrong with it?
```suggestion
    curl https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | gpg --dearmor -o /etc/apt/keyrings/cran-archive-keyring.gpg && \
```

Comment (floriansipp) [reply apps/rstudio/Dockerfile:21]: Probably nothing but i used the vscode example as you advised and it's probably overkill and can be simplified , testing now

File: apps/rstudio/Dockerfile (hunk 4 above)
Comment (floriansipp) [reply apps/rstudio/Dockerfile]: Indeed

File: apps/rstudio/Dockerfile (hunk 5 above)
Comment (floriansipp) [reply apps/rstudio/Dockerfile]: Steak is cooked, nothing wrong with that

**Review by greut — APPROVED** [review]

**Pull request merged**